src/ast/mod.rs

```diff
@@ -785,64 +669,18 @@ impl ToSexp for RunConfig {
     }
 }
 
-#[derive(Clone, Debug, PartialEq, Eq, Hash)]
-pub struct NormRunConfig {
-    pub ctx: CommandId,
-    pub ruleset: Symbol,
-    pub until: Option<Vec<NormFact>>,
-}
-
-impl NormRunConfig {
-    pub fn to_run_config(&self) -> RunConfig {
-        RunConfig {
-            ruleset: self.ruleset,
-            until: self
-                .until
-                .as_ref()
-                .map(|v| v.iter().map(|f| f.to_fact()).collect()),
-        }
-    }
-}
-
-/// A normalized function declaration- the desugared
-/// version of a [`FunctionDecl`].
-/// TODO so far only the merge action is normalized,
-/// not the default value or merge expression.
-#[derive(Clone, Debug, PartialEq, Eq, Hash)]
-pub struct NormFunctionDecl {
-    pub name: Symbol,
-    pub schema: Schema,
-    // todo desugar default, merge
-    pub default: Option<Expr>,
-    pub merge: Option<Expr>,
-    pub merge_action: Vec<NormAction>,
-    pub cost: Option<usize>,
-    pub unextractable: bool,
-}
-
-impl NormFunctionDecl {
-    pub fn to_fdecl(&self) -> FunctionDecl {
-        FunctionDecl {
-            name: self.name,
-            schema: self.schema.clone(),
-            default: self.default.clone(),
-            merge: self.merge.clone(),
-            merge_action: self.merge_action.iter().map(|a| a.to_action()).collect(),
-            cost: self.cost,
-            unextractable: self.unextractable,
-        }
-    }
-}
+pub type FunctionDecl = GenericFunctionDecl<Symbol, Symbol, ()>;
```

Comment:
Documentation (in particular, explain the difference between FunctionDecl and ResolvedFunctionDecl)